### ISSUE TITLE
Allow to specify generated package name for ProxyClass'es

### ISSUE THREAD

**Issue by tomkoptel** [realm/realm-java#3890]
#### Goal

Change the package name that is used to generate `ProxyClass`es. The reason is to hide a realm classes under private package modifier. I don't want to leak Realm models and let the be public classes. It would be great if I can hide generated classes inside the same package I have defined Realm models. In particular, example below you can see it be a part of `com.example` package. I can see it to be a part of `build.gradle.` script configuration where you can point the location of desired package name.

#### Code Sample

```groovy

realm {
  proxyPackageName "com.my.package"
}

```

<img width="544" alt="screen shot 2016-12-09 at 13 39 07" src="https://cloud.githubusercontent.com/assets/758909/21049504/82488774-be15-11e6-9252-0614d1fd2f24.png">

<img width="401" alt="screen shot 2016-12-09 at 13 39 27" src="https://cloud.githubusercontent.com/assets/758909/21049512/860fe24e-be15-11e6-95d1-8196b1f1cc8b.png">

<img width="586" alt="screen shot 2016-12-09 at 13 46 11" src="https://cloud.githubusercontent.com/assets/758909/21049561/dda84582-be15-11e6-8fe4-2c80e87543bd.png">

**Comment by Zhuinden** [comment]
Doesn't the generated RealmProxy contain a bunch of things package-visible only from `io.realm`?

**Comment by tomkoptel** [comment]
@Zhuinden yes, you are correct! So this leads to answer that this impossible to do unless you expose realm inner implementation ;(

<img width="530" alt="screen shot 2016-12-09 at 14 55 00" src="https://cloud.githubusercontent.com/assets/758909/21051223/8770fcfe-be1f-11e6-8050-305025c85345.png">
<img width="778" alt="screen shot 2016-12-09 at 14 54 30" src="https://cloud.githubusercontent.com/assets/758909/21051229/8ec78a86-be1f-11e6-9248-6f9976cbf04a.png">
<img width="785" alt="screen shot 2016-12-09 at 14 54 49" src="https://cloud.githubusercontent.com/assets/758909/21051227/8c15622c-be1f-11e6-841e-0bf10e9f41d7.png">

**Comment by zaki50** [comment]
How do you hide Realm classes such as `io.realm.Realm`?
